lib/pbench/server/api/resources/users_api.py

```diff
@@ -282,23 +297,52 @@ def post(self):
                         "message": "failure message"
                     }
         """
-        auth_token = self.auth.get_auth_token(self.logger)
-        user = Auth.verify_auth(auth_token=auth_token)
+        auth_token = Auth.get_auth_token()
+        state = Auth.verify_token_only(auth_token)
 
-        # "None" user represents that either the token is not present in our database or it is an expired token.
-        # Expired token is already deleted by now if we reach here.
-        # In either case we will return suceess to a client, since the same token can not be used again.
+        # The only time a token won't be in the database is if the token is
+        # not a valid token.  Valid tokens that are not expired should be in
+        # the database, and expired tokens might or might not be in the
+        # database.  If the valid token is found in the database associated
+        # with user, remove it.
 
-        if user:
-            try:
-                ActiveTokens.delete(auth_token)
-            except Exception:
-                self.logger.exception("Exception occurred deleting an auth token")
-                abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
-            else:
-                self.logger.debug("User {} logged out", user.username)
+        if state == Auth.TokenState.INVALID:
+            current_app.logger.info("User logout with invalid token: {}", auth_token)
         else:
-            self.logger.info("User logout with invalid or expired token")
+            token = ActiveToken.query(auth_token)
+            if token:
+                user = token.user
+                if not user:
+                    current_app.logger.info(
+                        "Unknown user logged out with verified or expired token"
+                    )
+                else:
+                    try:
+                        ActiveToken.delete(auth_token)
+                    except Exception:
+                        current_app.logger.exception(
+                            "Exception occurred deleting auth token {!r} for user {!r}",
+                            auth_token,
+                            user.username,
+                        )
+                        if state == Auth.TokenState.VERIFIED:
+                            # Only report an internal error to the user if the token
+                            # has been verified.
+                            abort(
+                                HTTPStatus.INTERNAL_SERVER_ERROR,
+                                message="INTERNAL ERROR",
+                            )
```

Comment:
Agree, but I don't see how further changes to this PR are worth making given where we are going with KeyCloak.

Reply:
I think that there are two issues here.

The first is that, if we find a token in the database but cannot delete it because it's no longer there, then we should return `200`, not `500`.  (I think this is easily addressed.)

The second is that, if we incur some other database error, we don't do anything (unless the `state` is `VERIFIED`), and we fall through and return `OK`.  I don't see why the state of the token should play into what the Server returns in this situation -- it seems to me that it should either return `200` (because there is nothing for the user to do about the problem) or `500` (because it really was an internal server error).  And, addressing this makes the code simpler.

And, given that you have other changes to make to this PR, I think it's worth making this change, as well.